### ISSUE TITLE
windows_task not idempotent when task_name includes parent folder

### ISSUE THREAD

**Issue by idflyfish** [chef/chef#7291]
## Description

The windows_task not idempotent when task_name includes parent folder. The resource will converge every time as a result, even when properties have not changed.

## Chef Version

14.1.12

## Platform Version

Windows Server 2016

## Replication Case

1. Create a cookbook called windows_task_test.
2. Create a recipe called my_task.rb.
3. In recipes/my_task.rb configure a windows_task resource with a task_name that includeds a parent folder (see the example below).

  ```ruby
  windows_task 'MyTask Under MyOrg' do
    task_name '\MyOrg\MyTask'
    command 'powershell -NoLogo -NonInteractive -ExecutionPolicy Bypass "Write-EventLog -LogName Application -EventID 10000 -Source Chef -EntryType Information -Message MyTask_Under_MyOrg"'
    run_level :highest
    frequency :daily
    start_time '03:00'
    action :create
  end
  ```

4. Execute a chef-client run ensuring that the windows_task_test::my_task recipe is included in the runlist.

## Client Output

```
Starting Chef Client, version 14.1.12
[2018-05-24T11:17:44-06:00] INFO: *** Chef 14.1.12 ***
[2018-05-24T11:17:44-06:00] INFO: Platform: x64-mingw32
[2018-05-24T11:17:44-06:00] INFO: Chef-client pid: 2588
[2018-05-24T11:17:44-06:00] INFO: The plugin path C:\chef\ohai\plugins does not exist. Skipping...
Creating a new client identity for Administrator using the validator key.
[2018-05-24T11:17:47-06:00] INFO: Client key C:/ChefInstall/.chef/Administrator.pem is not present - registering
[2018-05-24T11:17:47-06:00] INFO: Setting the run_list to [#<Chef::RunList::RunListItem:0x0000000005bc0228 @version=nil, @type=:recipe, @name="windows_task_test::my_task">] from CLI options
[2018-05-24T11:17:47-06:00] INFO: Run List is [recipe[windows_task_test::my_task]]
[2018-05-24T11:17:47-06:00] INFO: Run List expands to [windows_task_test::my_task]
[2018-05-24T11:17:47-06:00] INFO: Starting Chef Run for Administrator
[2018-05-24T11:17:47-06:00] INFO: Running start handlers
[2018-05-24T11:17:47-06:00] INFO: Start handlers complete.
resolving cookbooks for run list: ["windows_task_test::my_task"]
[2018-05-24T11:17:48-06:00] INFO: Loading cookbooks [windows_task_test@0.0.0]
Synchronizing Cookbooks:
[2018-05-24T11:17:48-06:00] INFO: Storing updated cookbooks/windows_task_test/recipes/my_task.rb in the cache.
  - windows_task_test (0.0.0)
Installing Cookbook Gems:
Compiling Cookbooks...
Converging 1 resources
Recipe: windows_task_test::my_task
  * windows_task[MyTask Under MyOrg] action create[2018-05-24T11:17:48-06:00] INFO: Processing windows_task[MyTask Under MyOrg] action create (windows_task_test::my_task line 1)

    - windows_task[MyTask Under MyOrg] task created
[2018-05-24T11:17:48-06:00] INFO: Chef Run complete in 0.792092 seconds

Running handlers:
[2018-05-24T11:17:48-06:00] INFO: Running report handlers
Running handlers complete
[2018-05-24T11:17:48-06:00] INFO: Report handlers complete
Chef Client finished, 1/1 resources updated in 03 seconds
```

## Stacktrace

NA

**Comment by tas50** [comment]
This is fixed by #7371. Thanks for the report @idflyfish

**Comment by idflyfish** [comment]
Thanks everyone!
